src/mesh/coordinates.cxx

```diff
@@ -857,11 +857,12 @@ void Coordinates::setParallelTransform(Options* options) {
 }
 
 ParallelTransform& Coordinates::getParallelTransform() {
-  // Return a reference to the ParallelTransform object
+  if (transform == nullptr) {
+    throw BoutException(_("Coordinates instance does not have parallel transform set"));
```

Comment:
Is there a case for making this an `ASSERT1` to make `getParallelTransform()` more inline-able in fully optimized builds? Probably doesn't matter, because this `if` really shouldn't ever be a significant overhead as `getParallelTransform` shouldn't ever be in an inner loop...

Reply:
Given how simple this function is, I've moved it into the header so that it could be inlined.